low-level/message_handlers/node_data_msg_handler.py

```diff
@@ -742,144 +737,75 @@ def _send_ifdata_json_msg(self, sensor_type, resource_id, resource_type, state,
-                        identified_cables[interface_name] = self.FAULT_RESOLVED
-
-                        if self.interface_fault_state and interface_name in self.interface_fault_state:
-                            # After the cable fault is resolved, unset the flag for interface
-                            # So that, it can be tracked further for any failure
-                            self.INTERFACE_FAULT_DETECTED = False
-                            self.interface_fault_state[interface_name] = self.INTERFACE_FAULT_DETECTED
-
-                            # Also clear the global nw interface dictionary
-                            self.prev_nw_status[interface_name] = phy_link_status
-
-                    self.prev_cable_cnxns[interface_name] = phy_link_status
-            else:
-                logger.debug(f"Cable connection state is unknown with '{interface_name}'")
-
-        return identified_cables
+            if_name = interface["ifId"]
+            nw_if_status = interface["nwStatus"].upper()
+            # if nw_fault
+            if nw_if_status in ["DOWN", "UNKNOWN"]:
+                nw_cable_status = interface["nwCableConnStatus"].upper()
+                # if nw_cable_disconneted
+                if nw_cable_status == "DISCONNECTED":
+                    if self.prev_cable_status.get(if_name) != nw_cable_status:
+                        # raise nw_cable_fault_alert
+                        self.prev_cable_status[if_name] = nw_cable_status
+                        self._send_ifdata_json_msg(
+                            if_name, self.NW_CABLE_RESOURCE_TYPE, self.FAULT,
+                            cable_alert_desc.format(if_name, nw_cable_status))
+                # elif nw_cable_connected
+                elif nw_cable_status == "CONNECTED":
+                    if self.prev_cable_status.get(if_name) == "DISCONNECTED":
+                        # raise nw_cable_fault_resolved_alert
+                        self.prev_cable_status[if_name] = nw_cable_status
+                        self._send_ifdata_json_msg(
+                            if_name, self.NW_CABLE_RESOURCE_TYPE,
+                            self.FAULT_RESOLVED,
+                            cable_alert_desc.format(if_name, nw_cable_status))
+                    if self.prev_nw_if_status.get(if_name) != "DOWN":
+                        # raise nw_fault_alert
+                        self.prev_nw_if_status[if_name] = "DOWN"
+                        self._send_ifdata_json_msg(
+                            if_name, self.NW_RESOURCE_TYPE, self.FAULT,
+                            nw_alert_desc.format(if_name, "DOWN"))
+                # nw_cable_state is unknown and nw_is_faulty
+                else:
+                    if self.prev_nw_if_status.get(if_name) != "DOWN":
+                        # raise nw_fault_alert
+                        self.prev_nw_if_status[if_name] = "DOWN"

```

Comment:
DOWN or UNKNOWN

Reply:
`prev_nw_if_status` is value set by us to check if the prev status and for caching purpose it is set to "DOWN"/"UP" only @cdeshmukh , so UNKNOWN for nw_interface_status will be considered as "DOWN" and the variable will have "DOWN" value.